projects/Douglas Mentoring/day1.py

```diff
@@ -0,0 +1,92 @@
+import getpass
+import netmiko
+# json was here bc as I'm using textfsm to pull data from cisco devices
+# it comes in json i needed to show the output before deciding how to tread the date
+import json
+import csv
+
+# Where i'm pulling the devices list / Soon I'll make this automated which will pull direct from NARF
+file = open(r'/home/segattod/segattod-workspace/src/Segattod-package/myproject/Light_level_checks/list.txt',
+            'r')
+file = file.read().splitlines()
+
+username = input('User Name:')
+password = getpass.getpass()
+
+# Variable for "append" the data later
+formatted_data = []
+
+# for now just to name the file / Soon to add for user to add input :)
+whid = 'YEG1'
+
+
+# Function to sorte the data
+def light_level_check(connection):
+    # Command to checks light level per interface
+    transceiver_output = connection.send_command('show interface transceiver', read_timeout=90,
+                                                 use_textfsm='~/ntc-templates/ntc_templates/templates/cisco_ios_show_interface_transceiver.textfsm')
+
+    for x in transceiver_output:
+        # added both variables below to do a check at that moment based on the "output" for that device interface the loop is going through
+        # why? bc was the way i could think to append(look below) all data to export later in CSV
+        # basically i use the "entry" information from "x" do all the checks and move to the next, and so on.
+        sfpmodel = connection.send_command(f'sh idprom interface {x["iface"]} detail | i Vendor Name')
+        sfpspeed = connection.send_command(
+            f'sh idprom interface {x["iface"]} detail | i Administrative Speed')
+
+        # Checking if that entry maches with 'cisco'
+        if 'CISCO' in sfpmodel.upper():
+            sfpM = 'Cisco'
+        else:
+            sfpM = 'Non-Cisco'
+        # Checking if that entry has a match with 10000, then with 1000
+        if '10000' in sfpspeed:
+            sfpS = '10GB'
+        elif '1000' in sfpspeed:
+            sfpS = '1GB'
+        else:
+            sfpS = 'Unknown'
+        # checking if the light level for fiber connections are acceptable
+        # AND add all that compiled data to "formatted_data"
+        ## I know I could have used "return" but when i did it it was not "appending"
+        ## it was adding all for one device and erasing and add the new ones, should be simple mistake from my end but i could not figure lol
+        if x['rx_pwr'] == 'N/A':
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'No Connection', sfpM, sfpS))
+        elif float(x['rx_pwr']) >= -10:
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'Pass', sfpM, sfpS))
+        else:
+            formatted_data.append((host, x['iface'], x["rx_pwr"], 'Fail', sfpM, sfpS))
```

Comment:
Stay DRY:
```
        pwr = x['rx_pwr']
        iface = x['iface']
        if pwr == 'N/A':
            formatted_data.append((host, iface, pwr, 'No Connection', sfpM, sfpS))
        elif float(x['rx_pwr']) >= -10:
            formatted_data.append((host, iface, pwr, 'Pass', sfpM, sfpS))
        else:
            formatted_data.append((host, iface, pwr, 'Fail', sfpM, sfpS))
```

BUT that can be simplified even more using try/except as control logic:

```suggestion
        pwr: str = sfp['rx_pwr']
        try:
            state = 'Pass' if float(pwr) >= MIN_LIGHT_LEVEL else 'Fail'
        except ValueError:  # Could not convert to a float
            state = 'No Connection'

        formatted_data.append((host, (sfp['iface']), pwr, state, model, speed))
```